### DISCUSSION TITLE
TRIM question.

### DISCUSSION THREAD

**Discussion by BerserkeR-Git** [007revad/Synology_HDD_db#337] [Q&A] [answered]
I used your wonderful script to add my M.2's to the compatibility list and I've successfully created a new storage pool.
But I saw in your other scripts an option to enable 'TRIM' for the SSD's, which I don't get using the 'Synology_HDD_db' script.

I'm planning to run my docker containers on these M.2's so I guess I'll need TRIM, right? How do I go about enabling it or is it already configured correctly like this?

Thanks in advance.

**Comment by BerserkeR-Git** [comment]
I have done some reading and it's suggested to add 'discard' under your ssd volume in the file 'fstab' at `/etc/fstab/`

original:
`/dev/mapper/cachedev_1 /volume2 btrfs auto_reclaim_space,ssd,synoacl,relatime,nodev 0 0`
edited:
`/dev/mapper/cachedev_1 /volume2 btrfs auto_reclaim_space,ssd,synoacl,relatime,nodev,discard 0 0`

Can someone confirm this is the correct way to go about this?

**Comment by 007revad** [comment, marked as answer]
None of my scripts have an option to enable TRIM. But one of them does explain how to enable TRIM.

To enable and schedule TRIM in Storage Manager go to: 
1.  Storage Pool.
2. Click on the SSD or NVMe Storage Pool.
3. Click the `...`.
4. Click Settings.
5. Click Enable TRIM.
6. Set a schedule.
7. Click Save.

**Reply by BerserkeR-Git** [reply]
Understood. Do you have a personal recommendation on how frequently to run TRIM?

Just to put my mind at ease, have you ever seen the suggestion I posted above? If so, do you know if it would work properly as well?

**Comment by 007revad** [comment]
I have no idea if editing fstab works in DSM 7. It used to work in older DSM versions. 

I prefer to use Synology's way to enable TRIM in DSM. I always leave the TRIM schedule set to the default once a day.

![image](https://github.com/user-attachments/assets/058b4a72-aa31-4fcb-9924-6b871d75019e)

**Reply by BerserkeR-Git** [reply]
I want to thank you for answering my questions so quickly and again for your work. I'll revert the changes in `/etc/fstab` and use the TRIM option in DSM (I didn't know about it before). I'll also mark the correct answer, **thank you!**

> When discard is enabled in `/etc/fstab`, TRIM commands are sent to the SSD automatically during regular file operations. This can be effective for real-time management of free space but might lead to increased wear if not properly managed.

TRIM scheduling in DSM seems to be the best option.
